README.md

```diff
@@ -1,2 +1,51 @@
 # gs-plugin
 Dockstore Google Cloud Storage file provisioning plugin
+
+The gs-plugin enables download, upload, and setting metadata on uploaded objects for
+Google Cloud Storage (GCS) on the Google Cloud Platform (GCP) through the Google Cloud Storage Client library 
+
+To allow gs-plugin to run the Google Cloud Storage Client library you must setup authentication;
+follow the instructions for setting up authentication as described in 
+https://cloud.google.com/storage/docs/reference/libraries. 
+
+Once authentication is setup the gs-plugin can download and upload files
+when it detects a path pointing to a GCS location, e.g. gs://bucket/path_to_file
+
+### Note
+The above instructions describe setting up a service account for authentication; 
+however the gs-plugin can also obtain authentication through a user account
+as described here https://cloud.google.com/sdk/docs/authorizing#types_of_accounts. 
+
+To optionally setup user account authentication you can install the Google Cloud SDK and use
+the gcloud utility from the command line. Instructions for SDK installation and other topics 
+can be found here: https://cloud.google.com/sdk/docs/how-to
+
+
+## Usage
+
+Here is an example input JSON file to a tool:
+```
+$ cat test.gs.json
+{
+  "input_file": {
+        "class": "File",
+        "path": "gs://genomics-public-data/references/GRCh38/chr1.fa.gz"
+          },
+    "output_file": {
+        "class": "File",
+        "metadata": "eyJvbmUiOiJ3b24iLCJ0d28iOiJ0d28ifQ==",
+        "path": "gs://my_bucket/bamstats_report.zip"
+    }
+}
+```
+Note that metadata is Base64 encoded in the JSON and creates metadata tags on the uploaded file.
```

Comment:
I realize this line is copied from the s3 plugin, but I just noticed it. I find "in the JSON" confusing; it sounds like it could be talking about the `test.gs.json` file, but it also needs to be clear that the value itself is a Base64-encoded JSON. 

It might also be nice to have an example JSON, and how you BASE64 encode it, but that may be overkill until people start using this feature.

Reply:
There are a couple examples here that can be copied/modified
https://github.com/ga4gh/dockstore/search?l=JSON&q=metadata

Reply:
Provided an example of encoding the metadata and reworded the Note